Build fails on MacOS

If I try and build using the Xcode generator (i.e. `cmake -G Xcode ..`) it configures OK, but when invoking `xcodebuild` I get

```
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild

note: Using new build system
note: Planning build
note: Constructing build description
Build system information
error: Multiple commands produce '<PbxCp /tmp/x/demos/shooter/RelWithDebInfo/shooter.app/Contents/Resources/data/TITLE_MUSIC.ogg>':
1) Target 'shooter' (project 'ALLEGRO') has copy command from '/Users/peterhull/Projects/allegro5/demos/shooter/data/TITLE_MUSIC.ogg' to '/tmp/x/demos/shooter/RelWithDebInfo/shooter.app/Contents/Resources/data/TITLE_MUSIC.ogg'
2) Target 'shooter' (project 'ALLEGRO') has copy command from '/tmp/x/demos/shooter/data/TITLE_MUSIC.ogg' to '/tmp/x/demos/shooter/RelWithDebInfo/shooter.app/Contents/Resources/data/TITLE_MUSIC.ogg'
```
(and similar errors for other resource files in shooter)

With Ninja it won't even configure
```
CMake Error:
  Running

   '/usr/local/bin/ninja' '-C' '/tmp/n' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:5598: multiple rules generate demos/shooter/shooter.app/Contents/Resources/data/DejaVuSans.ttf
```
Using allegro current master (0aeca8c6a72205dd4893799face0502720d1b341), cmake version 3.25.1, Xcode 12.4, MacOS 10.15.7

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,9 @@ string(REPLACE "." "" ALLEGRO_DLL_SHORTVER ${ALLEGRO_SOVERSION})
 # Search in the `cmake' directory for additional CMake modules.
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 
+# Reject in-source builds
+include(PreventInSourceBuilds)
+
 # Search in `deps' directories for dependency files.
 file(GLOB deps_subdirs
     "${PROJECT_SOURCE_DIR}/deps"

--- a/README.txt
+++ b/README.txt
@@ -41,7 +41,7 @@ Requirements
 We assume you have C and C++ compilers installed and functioning.
 We support gcc, clang and MSVC.
 
-Allegro also requires CMake 2.8.5 or later to build.
+Allegro also requires CMake 3.0 or later to build.
 You may download it from <http://www.cmake.org/>
 
 
@@ -165,6 +165,16 @@ cmake will detect your compiler setup and find the libraries which are
 installed on your system.  At the same time, you may select options to
 customise your build.  If you are unsure of what you are doing, leave all the
 options at the defaults.
+
+You must configure Allegro with a separate build directory. For example,
+
+    mkdir build
+    cd build
+    cmake ..
+
+If you configure Allegro to build in the source directory (i.e. `cmake .`)
+you will get an error message. Delete `CMakeCache.txt` and the `CMakeFiles`
+directory and re-configure as described above.
 
 Once the configuration step is successful, you will invoke another tool to
 build Allegro.  The tool depends on your compiler, but is usually either

--- a/cmake/PreventInSourceBuilds.cmake
+++ b/cmake/PreventInSourceBuilds.cmake
@@ -1,0 +1,17 @@
+# This function will prevent in-source builds
+# Based on https://github.com/InsightSoftwareConsortium/ITK/blob/1b5da45dc706e6d6803b52740fa50e0e5e7705e9/CMake/PreventInSourceBuilds.cmake
+function(PreventInSourceBuilds)
+  # make sure the user doesn't play dirty with symlinks
+  get_filename_component(srcdir "${CMAKE_SOURCE_DIR}" REALPATH)
+  get_filename_component(bindir "${CMAKE_BINARY_DIR}" REALPATH)
+
+  # disallow in-source builds
+  if("${srcdir}" STREQUAL "${bindir}")
+    message(FATAL_ERROR
+      "Allegro must not be configured to build in the source directory.\n"
+      "Please refer to README.md\n"
+      "Quitting configuration step")
+  endif()
+endfunction()
+
+PreventInSourceBuilds()

